File: schemas/src/main/java/com/scottlogic/deg/schemas/v3/Rule.java
package com.scottlogic.deg.schemas.v3;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.Collection;

@JsonDeserialize(using = RuleDeserializer.class)
@JsonSerialize(using = RuleSerializer.class)
public class Rule {
    public String description;
    public Collection<Constraint> constraints;

    public Rule() {}
}

File: profiler/src/main/scala/com/scottlogic/deg/models/Rule.scala
package com.scottlogic.deg.models

class Rule() {
  var rule: String = null
  var Constraints: List[IConstraint] = null

  def this(description: String, constraints: List[IConstraint]) {
    this()
    this.rule = description
    this.Constraints = constraints
  }
}